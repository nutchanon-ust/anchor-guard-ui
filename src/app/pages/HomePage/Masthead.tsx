import * as React from 'react';
import styled from 'styled-components/macro';
import { Logos } from './Logos';
import { Title } from './components/Title';
import { Lead } from './components/Lead';
import { A } from 'app/components/A';

export function Masthead() {
  return (
    <Wrapper>
      <Logos />
      <Title>React Boilerplate meets CRA</Title>
      <Lead>
        Now you can use the{' '}
        <A
          href="https://www.reactboilerplate.com/"
          target="_blank"
          rel="noopener noreferrer"
        >
          React Boilerplate
        </A>{' '}
        as a{' '}
        <A
          href="https://github.com/facebook/create-react-app"
          target="_blank"
          rel="noopener noreferrer"
        >
          Create React App
        </A>{' '}
        template.
      </Lead>
    </Wrapper>
  );
}

const Wrapper = styled.main`
  height: 60vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  min-height: 320px;
`;
