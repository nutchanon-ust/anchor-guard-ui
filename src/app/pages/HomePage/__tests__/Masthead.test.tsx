import * as React from 'react';
import { Masthead } from '../Masthead';
import { createRenderer } from 'react-test-renderer/shallow';

const shallowRenderer = createRenderer();

describe('<Masthead />', () => {
  it('should render and match the snapshot', () => {
    shallowRenderer.render(<Masthead />);
    const renderedOutput = shallowRenderer.getRenderOutput();
    expect(renderedOutput).toMatchSnapshot();
  });
});
