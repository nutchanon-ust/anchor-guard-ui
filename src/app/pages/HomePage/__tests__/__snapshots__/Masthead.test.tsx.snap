// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Masthead /> should render and match the snapshot 1`] = `
<styled.main>
  <Logos />
  <styled.h1>
    React Boilerplate meets CRA
  </styled.h1>
  <styled.p>
    Now you can use the
     
    <styled.a
      href="https://www.reactboilerplate.com/"
      rel="noopener noreferrer"
      target="_blank"
    >
      React Boilerplate
    </styled.a>
     
    as a
     
    <styled.a
      href="https://github.com/facebook/create-react-app"
      rel="noopener noreferrer"
      target="_blank"
    >
      Create React App
    </styled.a>
     
    template.
  </styled.p>
</styled.main>
`;
